### PR TITLE
Getting started guide

### PR DESCRIPTION
**Pull request by otrosien**
Closes Issue #23

### CODE REVIEW

File: docs/elasticsearch-cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  namespace: es-operator-demo
+data:
+  elasticsearch.yml: |
+    cluster.name: es-cluster
+    network.host: "0.0.0.0"
+    bootstrap.memory_lock: false
+    discovery.zen.ping.unicast.hosts: es-cluster
+    discovery.zen.minimum_master_nodes: 1
+---
+apiVersion: apps/v1beta1
```
Comment (mikkeloscar) [inline docs/elasticsearch-cluster.yaml]: `apps/v1`

File: docs/GETTING_STARTED.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Step 5 - Add Elasticsearch Data Sets
+
+Finally, let's add data nodes. There are two manifests for the purpose of this demo. The simple stack will launch one data node, and has auto-scaling off. There is also a scaling stack in case you want to experiment with auto-scaling settings.
+
+```
+kubectl apply -f elasticsearchdataset-simple.yaml
+kubectl apply -f elasticsearchdataset-scaling.yaml
+```
+
```
Comment (mikkeloscar) [inline docs/GETTING_STARTED.md]: I would add a step of checking that the expected resources are deployed. e.g. `kubectl get all` and a bit of description of the different parts.

File: docs/elasticsearch-cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        - name: node.master
+          value: "true"
+        - name: node.data
+          value: "false"
+        readinessProbe:
+          httpGet:
+            scheme: HTTP
+            path: /_cluster/health?local=true
+            port: 9200
+          initialDelaySeconds: 5
```
Comment (GEverding) [inline docs/elasticsearch-cluster.yaml]: Hi, was looking at this to give it a try. Should this be something a little longer given you are starting up a JVM and ES?

File: docs/elasticsearch-cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    discovery.zen.ping.unicast.hosts: es-cluster
+    discovery.zen.minimum_master_nodes: 1
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: es-master
+  namespace: es-operator-demo
+  annotations:
+    downscaler/exclude: "true"
```
Comment (hjacobs) [inline docs/elasticsearch-cluster.yaml]: I think this [kube-downscaler](https://github.com/hjacobs/kube-downscaler) annotation is not relevant for this tutorial, remove?

File: docs/es-operator.yaml [hunk 5]
```diff
@@ -1,10 +1,24 @@
...
 apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: es-operator
+  namespace: es-operator-demo
   labels:
     application: es-operator
     version: latest
+  annotations:
+    downscaler/exclude: "true"
```
Comment (hjacobs) [inline docs/es-operator.yaml]: Same here

File: docs/elasticsearch-cluster.yaml (hunk 3 above)
Comment (mikkeloscar) [reply docs/elasticsearch-cluster.yaml]: Yeah, probably we could give it 30 sec or so without a problem. Don't think we have really measured how fast it actually starts with these settings.

Comment (otrosien) [reply docs/elasticsearch-cluster.yaml]: I increased it to 20s. It doesn't make a big difference because it's for the readiness probe only.

**Comment by thomasdraebing** [comment]
Hi,

I just found this project and naively tried to set up the demo using the Getting Started-guide from this pull request. When I do so, I get two data-nodes after applying the CR, but one (id 0 of the StatefulSet) terminates as soon as it is ready, without going into a "Failed"-state. This is then happening in a loop. This happened for both the simple and the scaling example. 

I haven't had the chance to figure out what exactly goes wrong yet. If I should find the cause I will comment on that :-)

**Comment by mikkeloscar** [comment]
:+1:

**Comment by otrosien** [comment]
@thomasdraebing I'm still ironing out the last bugs in the manifests. Also, even for just getting started  the resource consumption for Elasticsearch is pretty high. Make sure you assign enough CPU and memory for your cluster.

**Comment by otrosien** [comment]
todo: make failing the documentation end-to-end tests a build failure.

File: docs/es-operator.yaml [hunk 6]
```diff
@@ -1,7 +1,19 @@
+apiVersion: v1
```
Comment (mikkeloscar) [inline docs/es-operator.yaml]: This file is referenced in the README. Link should be updated there.

File: docs/GETTING_STARTED.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+We differentiated the stacks using an Elasticsearch node tag called `group`. It is advised to use this tag to bind indices with the same scaling requirements to nodes with the same `group` tag, by using the shard allocation setting like this:
+
+ ```
+curl -XPUT localhost:9200/demo-index -HContent-type:application/json \
+ -d '{"number_of_shards":5, "number_of_replicas":2, "routing.allocation.include.group": "group2"}'
+ ```
+
+## Advanced Step: Auto-Scaling
+
+Once you have gathered some experience in how the ES Operator handles your data nodes, you can start experimenting with auto-scaling features. The README.md offers some examples of different scaling scenarios, or look at the manifests of our [demo at the microXchg 2019](https://github.bus.zalan.do/otrosien/microxchg-demo).
```
Comment (njuettner) [inline docs/GETTING_STARTED.md]: I guess this is wrong, it points to an Internal repo

File: .travis.yml [hunk 8]
```diff
@@ -7,12 +7,23 @@ go:
...
+        - go get github.com/lawrencewoodman/roveralls
+      script:
+        - make test
+        - make build.docker
+        - roveralls -ignore cmd/e2e
+        - goveralls -v -coverprofile=roveralls.coverprofile -service=travis-ci
+    - stage: Documentation Tests
+      before_script:
+        - go get sigs.k8s.io/kind
+        - curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.12.0/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
```
Comment (njuettner) [inline .travis.yml]: I would use an env for changing Kubernetes version

Comment (otrosien) [reply .travis.yml]: hm.. we're not re-using this anywhere except at this place. Once we do, I'll externalize this into env.

File: docs/GETTING_STARTED.md (hunk 7 above)
Comment (otrosien) [reply docs/GETTING_STARTED.md]: thx. updating the PR

**Comment by njuettner** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:

**Comment by adichad** [comment]
in some scenarios, ClusterRole definitions aren't permitted. do namespace-level Role definitions suffice? I would guess so. maybe that will in the getting started documentation and folder we could add an example "roles.yaml" (parallel to cluster-roles.yaml)? that only requests namespace-level permissions? 

on the other hand, a cluster i have at hand rejecting the cluster-roles.yaml from being applied force me to read up a bit more about k8s ;)

**Pull request merged**